lazysnap/src/main/java/com/google/accompanist/lazysnap/SnappingFlingBehavior.kt

```diff
@@ -0,0 +1,689 @@
+        return when {
+            currentItemInfo.offset < snapOffset - (currentItemInfo.size / 2) -> {
+                (currentItemInfo.index + 1).coerceIn(0, lazyListState.layoutInfo.lastIndex)
+            }
+            else -> currentItemInfo.index
+        }
+    }
+
+    private suspend fun ScrollScope.performSpringFling(
+        initialItem: LazyListItemInfo,
+        targetIndex: Int,
+        initialVelocity: Float = 0f,
+    ): Float {
+        // If we're already at the target + snap offset, skip
+        if (initialItem.index == targetIndex &&
+            initialItem.offset == snapOffsetForItem(layoutInfo, initialItem)
+        ) {
+            Napier.d(
+                message = {
+                    "Skipping spring: already at target. " +
+                        "vel:$initialVelocity, " +
+                        "current item: ${initialItem.log()}, " +
+                        "target: $targetIndex"
+                }
+            )
+            return initialVelocity
+        }
+
+        Napier.d(
+            message = {
+                "Performing spring. " +
+                    "vel:$initialVelocity, " +
+                    "current item: ${initialItem.log()}, " +
+                    "target: $targetIndex"
+            }
+        )
+
+        val itemSpacing = lazyListState.layoutInfo.itemSpacing
+        var velocityLeft = initialVelocity
+        var lastValue = 0f
+
+        try {
+            // Update the animationTarget
+            animationTarget = targetIndex
+
+            AnimationState(
+                initialValue = 0f,
+                initialVelocity = initialVelocity,
+            ).animateTo(
+                targetValue = when {
+                    targetIndex > initialItem.index -> initialItem.size + itemSpacing
+                    else -> -(initialItem.size + itemSpacing)
+                }.toFloat(),
+                animationSpec = snapAnimationSpec,
+            ) {
+                val delta = value - lastValue
+                val consumed = scrollBy(delta)
+                lastValue = value
+                velocityLeft = velocity
+
+                if (checkSnapBack(initialVelocity, targetIndex, ::scrollBy)) {
+                    cancelAnimation()
+                } else if (abs(delta - consumed) > 0.5f) {
+                    // If we're still running but some of the scroll was not consumed,
+                    // cancel the animation now
+                    cancelAnimation()
+                }
+            }
+        } finally {
+            animationTarget = null
+        }
+
+        Napier.d(
+            message = {
+                "Spring fling finished. Distance: $lastValue. Final vel: $velocityLeft"
+            }
+        )
+
+        return velocityLeft
+    }
+
+    /**
+     * Returns true if we needed to perform a snap back, and the animation should be cancelled.
+     */
+    private inline fun AnimationScope<Float, AnimationVector1D>.checkSnapBack(
```

Comment:
probably rename to something like snapBackIsNeeded()

Reply:
Done.